core/src/main/java/org/apache/brooklyn/core/catalog/internal/CatalogBomScanner.java

```diff
@@ -195,22 +210,29 @@ public void removedBundle(Bundle bundle, BundleEvent bundleEvent, Iterable<? ext
             for (CatalogItem<?, ?> item : items) {
                 LOG.debug("Unloading {} {} from catalog", item.getSymbolicName(), item.getVersion());
 
-                removeFromCatalog(item);
+                catalogBundleLoader.removeFromCatalog(item);
             }
         }
+    }
 
-        private void removeFromCatalog(CatalogItem<?, ?> item) {
-            try {
-                getManagementContext().getCatalog().deleteCatalogItem(item.getSymbolicName(), item.getVersion());
-            } catch (Exception e) {
-                Exceptions.propagateIfFatal(e);
-                LOG.warn(Strings.join(new String[] {
-                    "Failed to remove", item.getSymbolicName(), item.getVersion(), "from catalog"
-                }, " "), e);
-            }
+    @Beta
+    public class CatalogBundleLoader {
```

Comment:
In CatalogBomScanner, now that this refactorization has being done we should remove the constructor 
```
CatalogPopulator(BundleContext bundleContext, ManagementContext managementContext)
``` 
which was introduced for this purpose, including deleting the `bundleContext` and `managementContext` members and re-simplifying any methods that use them.